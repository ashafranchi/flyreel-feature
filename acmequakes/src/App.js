import React, { useEffect, useState } from 'react';
import { Badge, Box, Grid } from "@chakra-ui/react";
import './App.css';
import axios from 'axios';
import { Map, Marker } from "google-maps-react";
import MapContainer from './MapContainer';


function App() {
  let url = "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/significant_month.geojson"
  const [data, setData] = useState([])

  async function fetchEarthquakes() {
    const response = await axios.get(url);
    setData(response.data.features)
    console.log(data);
}



  useEffect(() => {
    fetchEarthquakes()
  }, [])

  return (
    <div className="App">
      <header className="App-header">
        <div className="quakeContainer">
        <h1>Earthquakes from the past month: </h1>
      <Grid templateColumns="repeat(4, 1fr)" gap={6}>
      {data.map((e) => (
      <Box maxW="md" borderWidth="1px" borderRadius="lg" overflow="hidden">
      <Box p="6">
        <Box d="flex" alignItems="baseline">
          <Badge borderRadius="full" px="2" colorScheme="teal">
            Recent
          </Badge>
          <Box
            color="gray.500"
            fontWeight="semibold"
            letterSpacing="wide"
            fontSize="xs"
            textTransform="uppercase"
            ml="2"
          >
            {e.properties.alert}
          </Box>
        </Box>

        <Box
          mt="1"
          fontWeight="semibold"
          as="h4"
          lineHeight="tight"
        >
          {e.properties.title}
        </Box>

        <Box>
          {e.properties.felt}
          <Box className="Felt" as="span" color="blue.600" fontSize="md">
            people felt
          </Box>
        </Box>
      </Box>
    </Box>
    ))}
    </Grid>
    <MapContainer data={data}/>
  </div>
      </header>
    </div>
  );
}

export default App;
