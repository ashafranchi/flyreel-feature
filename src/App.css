.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #90CDF4;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-left: 1em;
  padding-right: 2em;
  color: white;
  padding-bottom: 2em;
  padding-top: 2em;
}

.App-footer {
  background-color: #90CDF4;
}

.App-link {
  color: #61dafb;
}

.Felt {
  margin-left: 6px;
  margin-bottom: 4px;
}

.Quake-container {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  padding-top: 1em;
  padding-right: 2em;
  padding-left: 2em;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
