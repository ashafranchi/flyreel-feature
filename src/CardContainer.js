import React, { useState, useEffect, useRef } from 'react';
import { Badge, Box, Grid, Heading } from "@chakra-ui/react";
import moment from 'moment'
import { createBreakpoints } from "@chakra-ui/theme-tools"

const breakpoints = createBreakpoints({
  sm: "30em",
  md: "48em",
  lg: "62em",
  xl: "80em",
  "2xl": "96em",
})

const CardContainer = ({ data }) => {

  return (
    <Grid templateColumns={{base: "repeat(2, 1fr)", md: "repeat(3, 1fr)", lg: "repeat(4, 1fr)"}} gap={6}>
      {data && data.map((e) => (
        <Box key={e.properties.ids} maxW="md" borderWidth="2px" borderRadius="lg" overflow="hidden">
          <Box p="6">
            <Box d="flex" alignItems="baseline">
              <Badge borderRadius="full" px="2" colorScheme="blue">
                {moment(e.properties.time).format('LL')}
              </Badge>
            </Box>
          <Box mt="1" paddingTop="0.5em" fontSize="xl" fontWeight="semibold" as="h4" lineHeight="tight">
            {e.properties.title}
          </Box>
          <Box paddingTop="0.5em" fontSize="lg" d="flex" justifyContent="center" alignItems="flex-end">
            {e.properties.felt}
            <Box marginBottom="3px" marginLeft={{base: "3px", sm: "6px"}} as="span" fontSize={{base: "12px", sm: "md"}}>
              affected
            </Box>
          </Box>
        </Box>
      </Box>
      ))}
    </Grid>
  )
}

export default CardContainer;