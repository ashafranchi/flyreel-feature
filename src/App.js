import React, { useEffect, useState } from 'react';
import { Badge, Box, Grid, Heading } from "@chakra-ui/react";
import './App.css';
import axios from 'axios';
import { Map, Marker } from "google-maps-react";
import MapContainer from './MapContainer';
import CardContainer from './CardContainer';


function App() {
  let url = "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/significant_month.geojson"
  const [data, setData] = useState([])

  async function fetchEarthquakes() {
    const response = await axios.get(url);
    setData(response.data.features)
    console.log(data);
}



  useEffect(() => {
    fetchEarthquakes()
  }, [])

  return (
    <div className="App">
      <header className="App-header">
        <Heading as="h2" size="lg" >ACME Tools</Heading>
        <Heading as="h2" size="lg" >Menu</Heading>
      </header>
      <div className="Quake-container">
        <Heading as="h2" size="2xl">Earthquake Tracker</Heading>
        <Heading as="h4" size="md" fontWeight="300" paddingTop="0.5em">Made with Live USGS Data</Heading>
        <Heading as="h3" size="xl" paddingTop="1.5em" paddingBottom="1em" isTruncated>Recent Quakes </Heading>
        <CardContainer data={data} />
        <Heading as="h2" size="xl" paddingBottom="0.5em" paddingTop="1.5em" isTruncated>Quake Map</Heading>
        <MapContainer data={data} />
      </div>
      <footer className="App-footer">
        <Heading as="h2" size="lg" color="white" paddingBottom="1em" paddingTop="1.5em" isTruncated>Â© 2021 ACME Corporation</Heading>
      </footer>
    </div>
  );
}

export default App;
