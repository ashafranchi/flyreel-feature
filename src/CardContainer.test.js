import "./setupTests"
import App from './App';
import CardContainer from './CardContainer';

import { shallow, mount, render } from 'enzyme';

describe("CardContainer", () => {
  it("CardContainer should render CardContainer", () => {
    const wrapper = shallow(<CardContainer />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});