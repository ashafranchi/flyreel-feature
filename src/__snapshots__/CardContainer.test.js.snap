// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer CardContainer should render CardContainer 1`] = `
Array [
  <Grid
    gap={6}
    templateColumns={
      Object {
        "base": "repeat(2, 1fr)",
        "lg": "repeat(4, 1fr)",
        "md": "repeat(3, 1fr)",
      }
    }
  />,
]
`;
